hightouch_blueprints/verify_sync.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import sys
+import requests
+import shipyard_utils as shipyard
+import errors
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--access-token', dest='access_token', required=True)
+    parser.add_argument('--sync-id', dest='sync_id', required=True)
+    parser.add_argument('--sync-run-id', dest='sync_run_id', required=False)
+    args = parser.parse_args()
+    return args
+
+
+
+def get_sync_status(sync_id, sync_run_id, access_token):
+    """
+    Gets the sync status from the Hightouch API
+    see: https://hightouch.io/docs/api-reference/#operation/ListSyncRuns
+    """
+    sync_api = f"https://api.hightouch.io/api/v1/syncs/{sync_id}/runs"
+    api_headers = {
+        'authorization': f"Bearer {access_token}",
+        'Content-Type': 'application/json'
+    }
+    params = {
+        "runId": sync_run_id
+    }
+    try:
+        sync_run_response = requests.get(sync_api,
+                                              params=params,
+                                              headers=api_headers)
+        
+        sync_status_code = sync_run_response.status_code
+        sync_run_json = sync_run_response.json()
+        # check if successful, if not return error message
+        if sync_status_code == requests.codes.ok:
+            return sync_run_json
+            
+        elif sync_status_code == 400: # Bad request
+            print("Sync status request failed due to Bad Request Error.")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+            
+        elif sync_status_code == 401: # Incorrect credentials 
+            print("Incorrect credentials, check your access token")
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
+            
+        elif sync_status_code == 422: # Invalid status query
+            print(f"Check status Validation failed: {sync_run_json['details']}")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+        
+        else:
+            print("Unknown Error when sending request")
+            sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+        
+    except Exception as e:
+        print(f"Sync status for Sync {sync_id}:{sync_run_id} failed due to: {e}")
+        sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+
+
+def handle_sync_run_data(sync_run_data):
+    """
+    Analyses sync run data to determine status and print sync run information
+    
+    Returns:
+        status_code: Exit Status code detailing sync status
+    """
+    run_id = sync_run_data['id']
+    status = sync_run_data['status']
```

Comment:
All of the sync_run_data references are wrong. Should be `sync_run_data['data'][0]['field_to_pull_in]`

Reply:
Or you could just limit the initial storage of `sync_run_data` to be `sync_run_data['data'][0]` so these would all be valid.